### PR TITLE
Release new demo app and SDK improvements.

### PR DESCRIPTION
**Pull request by alonso-cadenas**
## Description

Jira Ticket: CALL-3749

## Merge Checklist

| Q                        | A 
| ------------------------ | --------------
| Adds Documentation?      | :white_check_mark:
| Any Dependency Changes?  |
| Patch: Bug Fix?          |
| Minor: New Feature?      | :white_check_mark:
| Major: Breaking Change?  |

[BRAVE Checklist](https://github.com/HubSpot/calling-extensions-sdk/blob/master/SHIP_WITH_CARE.md)

- [x] I have read the BRAVE checklist and confirmed if the following is necessary.

| Q                              | A 
| ------------------------------ | --------------
| Backwards Compatible?          |:white_check_mark:
| Rollout/Rollback Plan?         | 
| Automated test coverage?       | Unit tests
| Verified that changes work?    |:white_check_mark:
| Expect Dependencies to Fail?   |:white_check_mark:

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Deploy demo apps to GH Pages
+concurrency: ci-${{ github.ref }}
+on:
+  workflow_dispatch:
+  push:
+    branches: [project-demo-v1]
```
Comment (alonso-cadenas) [inline .github/workflows/deploy.yml:6]: @esme, should this be changed to `master`?

File: .github/workflows/preview.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+name: Deploy PR previews for demo apps
+run-name: ${{ github.actor }} is deploying a PR
+concurrency: preview-${{ github.ref }}
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: ['*demo*']
```
Comment (alonso-cadenas) [inline .github/workflows/preview.yml:7]: Same here

Comment (esme) [reply .github/workflows/preview.yml:7]: This gives us a preview of the demo apps on any branch named `*demo*` so we could leave as is.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (alonso-cadenas) [reply .github/workflows/deploy.yml:6]: https://github.com/HubSpot/calling-extensions-sdk/pull/120

File: .github/workflows/preview.yml (hunk 2 above)
Comment (alonso-cadenas) [reply .github/workflows/preview.yml:7]: https://github.com/HubSpot/calling-extensions-sdk/pull/120

**Review by esme — APPROVED** [review]
🚀

**Pull request merged**